src/table.rs

```diff
@@ -75,6 +78,33 @@ impl Table {
     }
 }
 
+/// Open an iceberg table by uri
+pub async fn open(uri: impl AsRef<str>) -> Result<Table> {
```

Comment:
- I prefer to accept `uri: &str` for more simple to under stand API.
- And I prefer to provide API that called `Table::open`.